README.md

```diff
@@ -1,15 +1,157 @@
+Head [here](https://platform.cloud.coveo.com/docs?api=Platform) for a refresher or to get antiquated with 
+Coveo API, as many different calls using the API are used within this project.
+
+For general information regarding Zapier, follow this [link](https://zapier.com/developer/documentation/v2/) to their main
+documentation page. Since this is part of their CLI Interface, you may also find it useful to have [this](https://www.npmjs.com/package/zapier-platform-cli) or
+[this](https://zapier.github.io/zapier-platform-cli/cli.html) on hand.
+
+In order to get a better grasp of the requirements for the CLI as well as how it functions, follow the first section, the setup portion, of this [tutorial](https://zapier.com/developer/start/introduction).
+You can follow the entire tutorial if you wish, but do not clone the Coveo app if you do, clone their example app instead.
+
+Note: you will need your own Zapier account or the Coveo Zapier account information in order to login from the command line in the `Run Zapier Login` step in the tutorial. If you
+choose to login from another account other than Coveo's, you will need to change the contents of the config file to the dev platform as well as get the correct redirect_uri
+after pushing the app for the first time. Hence, it is suggested you login with the Coveo Zapier account. Contact whomever has it on hand.
+
+Once you have completed the set up and logged in, you'll want to clone the Coveo app, move into the directory where the app code was stored, and install it's components:
+
+```bash
+git clone https://github.com/coveo-labs/zapier.git
+cd zapier
+npm install
+```
+Now, you should have the app and all of it's dependencies installed in the directory. If the app failed to update, run the following commands to
+ensure the app is updated:
+
+```bash
+sudo npm install zapier-platform-cli
+sudo npm install zapier-platform-core
+```
+The basics of setting up the app should be complete at this point. If any problems arose, diagnose them from the errors in the command line
+and resolve them.
+
+At any point if you want to see the available zapier commands and what they do, run `zapier help` or for more info on specific commands run
+`zapier help [command]`.
+
+## Adding in the Coveo App's Specific Information
+
+Now you'll have to update any information that is specific to the Coveo App. First, navigate to the `config.json` file. You should find the following inside:
+
+```json
+{
+ "PLATFORM": "platform.cloud.coveo.com",
+ "PUSH": "push.cloud.coveo.com",
+ "REDIRECT_URI": "https://zapier.com/dashboard/auth/oauth/return/APP-ID/"
+}
+```
+
+From here, ensure that `PLATFORM` and `PUSH` are the values associated with the Prod platform for Coveo. If for any reason you wish to change these two
+to either dev or qa (solely for testing purposes only), DO NOT push the same version of the app when you do. See the section regarding updating the app as to why you must do this. 
```

Comment:
I like this Readme, very detailed.

The repo is public, so anything specific to Coveo should be kept for Coveo only. (info about the dev & qa platforms, or the Coveo Zapier account) 
We will create a page in our internal Confluence for that info.